### DISCUSSION TITLE
Raster tiles from file on android

### DISCUSSION THREAD

**Discussion by Vanosuperhero** [maplibre/maplibre-native#3062] [Q&A]
Hello everyone!) Is it possible to use `file://` to get raster tiles from folders z/x/y on Android?

I'm trying to add a raster layer to a map via RasterLayer(id, uri)
and pass the uri as 

`"file://"+LocalContext.current.getExternalFilesDir(null)?.path+"/tiles.json"
`

Inside tiles.json there is information about the boundaries and zoom levels, as well as the path to the tiles:

> "tiles": ["file:///storage/emulated/0/Android/data/project.name/files/{z}/{x}/{y}.png"]

But for some reason mapLibre does not get tiles from there. The only way out is to add a local server, but i would like to avoid it if possible and transfer the tiles directly.

**Comment by louwers** [comment]
That sounds like something that should work.

You could also try providing the needed files as assets and use `asset://`.

**Reply by Vanosuperhero** [reply]
i used json with url and it worked so the problem is not reading from file. Can mapLibre independently retrieve rasters from folders? Mb there is a example of it?
